### PR TITLE
Avoid linkifying trailing period

### LINKED ISSUE #26: Trailing punctuation is included in link (periods, question marks)
sindresorhus/refined-github#381 introduced adding links to URLs in code, which is a really cool feature!

However, it doesn't follow the same rules as Markdown in Markdown code, such as the case where a link has a period at the end (the link on https://github.com/palmerhq/tsdx/pull/138/files goes to `https://github.com/alexreardon/tiny-invariant.`, with the period):

Screen Shot 2019-08-15 at 17 12 18

I can think of two ways to deal with this off the top of my head (maybe there's a better / easier alternative):

- use a different regular expression in Markdown files that more closely matches Markdown semantics
- don't allow periods at the last character of a URL? I haven't seen many (any?) URLs like this, and I think the number of URLs to be fixed by this is probably a lot higher...

### LINKED ISSUE #7577: `linkify-code` trailing dot/point/`.` picked up as part of URL/link
### Description

When an URL appears at the end of a sentence, it is often followed by a `.` without space. GitHub correctly highlights the URL w/o the `.` in the code view and makes the correct link clickable in Preveiw/View mode. Even my IDE (VSCode) excludes the `.` from the URL.

But the `linkify-code` feature incorrectly includes `.` as part of the clickable URL in the code view

### How to replicate the issue + URL

Modify any file to include an URL with path params (`/`) that ends with `.` and check the PR/commit in GitHub. For example:

```md
This repo is https://github.com/refined-github/refined-github.
```

Edit, even just this code block shows the issue. ☝️ 

Here's an example https://github.com/python/typeshed/commit/2fb9c35ff93db0e8e7297cbb5860a7bb6aa5dfa6#diff-9c749476405e46c668093de1fa173e3569a04cbd97331800ac4dc2a67059168eR105

### Extension version

24.7.10

### Browser(s) used

Opera, FireFox

### PR DESCRIPTION
**Pull request by fregante**
Hackish but fixes a 5.5-year old issue:

- fixes https://github.com/sindresorhus/linkify-urls/issues/26
- fixes https://github.com/refined-github/refined-github/issues/7577

### CODE REVIEW

**Comment by karlhorky** [comment]
@fregante thanks for this, nice! 🙌 

Does this also handle question marks, which was part of #26?

**Comment by fregante** [comment]
No it doesn't. Fixing it would be easy though, PR welcome

File: index.js [hunk 1]
```diff
@@ -20,16 +20,25 @@ const parseValue = (value, href) => {
...
-		href, // eslint-disable-line no-dupe-keys -- Ensures it's not overwritten
-	},
-	...parseValue(options.value, href),
-});
+function linkify(href, options = {}) {
+	// The regex URL mistakenly includes a dot at the end of the URL
+	let trailingDot = '';
+	if (href.endsWith('.')) {
+		href = href.slice(0, -1);
+		trailingDot = '.';
```
Comment (fregante) [inline index.js:28]:
```JS
trailingDot = url.slice(-1);
``` 

Plus the condition update

**Comment by karlhorky** [comment]
> No it doesn't. Fixing it would be easy though, PR welcome

Something like this?

- https://github.com/sindresorhus/linkify-urls/pull/50

**Pull request merged**